File: frontend/src/api/agent.ts
import axios from '../api/axios'

export interface Agent {
  id: number
  name: string
  description?: string
  avatarUrl?: string
  systemPrompt: string
  provider: string
  providerAgentId?: string
  model?: string
  tools?: any[]
  ownerId?: number
  isPublic: boolean
}

export async function getAgents(): Promise<Agent[]> {
  return axios.get('/agents') as unknown as Agent[]
}

export async function createAgent(agent: Partial<Agent>): Promise<Agent> {
  return axios.post('/agents', agent) as unknown as Agent
}

File: backend/src/main/java/com/agenthub/model/entity/Agent.java
package com.agenthub.model.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("agent")
public class Agent {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;

    private String description;

    private String avatarUrl;

    private String systemPrompt;

    private String provider; // minimax/coze/custom

    private String providerAgentId;

    private String model;

    private String tools; // JSON string

    private Long ownerId;

    private Boolean isPublic;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}
